From 742f564ee80749e9f1f3363092775545e37c0f87 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 9 May 2024 17:02:09 -0700
Subject: [PATCH] packet++: Initialize m_ExternalRawData in IDnsResource
 constructor

Fixes following warning with GCC-14

Packet++/src/DnsResource.cpp:29:24: error: '*this.pcpp::IDnsResource::m_ExternalRawData' may be used uninitialized [-Werror=maybe-uninitialized]
   29 |                 return m_ExternalRawData;
      |                        ^~~~~~~~~~~~~~~~~

Upstream-Status: Submitted [https://github.com/seladb/PcapPlusPlus/pull/1391]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 Packet++/src/DnsResource.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Packet++/src/DnsResource.cpp b/Packet++/src/DnsResource.cpp
index fec1bf05..c1f97041 100644
--- a/Packet++/src/DnsResource.cpp
+++ b/Packet++/src/DnsResource.cpp
@@ -10,7 +10,7 @@ namespace pcpp
 {
 
 IDnsResource::IDnsResource(DnsLayer* dnsLayer, size_t offsetInLayer)
-	: m_DnsLayer(dnsLayer), m_OffsetInLayer(offsetInLayer), m_NextResource(nullptr)
+	: m_DnsLayer(dnsLayer), m_OffsetInLayer(offsetInLayer), m_NextResource(nullptr), m_ExternalRawData(nullptr)
 {
 	char decodedName[4096];
 	m_NameLength = decodeName((const char*)getRawData(), decodedName);
-- 
2.45.0

